datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/DefaultMatchMaker.scala

```diff
@@ -677,8 +677,18 @@ object SelectSelectGroupbyChildDelta extends DefaultMatchPattern with PredicateH
               val aliasMap_exp = AttributeMap(
                 gb_2c.outputList.collect {
                   case a: Alias => (a.toAttribute, a) })
+
+              // avoid to transform a expression more than twrice
+              // accept for select and having
+              val transformedExpFlags = scala.collection.mutable.Map[ExprId, Int]()
+              aliasMap_exp.keySet.map(alias =>
+                transformedExpFlags += (alias.exprId -> 0))
               val sel_3q_exp = sel_3q.transformExpressions({
-                case attr: Attribute if aliasMap_exp.contains(attr) => aliasMap_exp(attr)
+                case attr: Attribute if aliasMap_exp.contains(attr) &&
+                  transformedExpFlags(attr.exprId) < 2 => {
+                  transformedExpFlags(attr.exprId) += 1
+                  aliasMap_exp(attr)
+                }
```

Comment:
In this way, we may not able to solve the complete issue as attributes may repeat in input/output list of select plan. Better create a wrapper and do the transform as below.

First create a wrapper class for alias
```
case class AliasWrapper(alias: Alias) extends UnaryExpression {
                override def child: Expression = null

                override protected def doGenCode(ctx: CodegenContext,
                    ev: ExprCode): ExprCode = ev

                override def dataType: DataType = alias.dataType
              }
```
And change the code as below.
```
 val aliasMap_exp = AttributeMap(
                gb_2c.outputList.collect {
                  case a: Alias => (a.toAttribute, AliasWrapper(a)) })
              val sel_3q_exp = sel_3q.transformExpressions({
                case attr: Attribute if aliasMap_exp.contains(attr) => aliasMap_exp(attr)
              }).transformExpressions {
                case AliasWrapper(alias: Alias) => alias
              }
```

Reply:
@ravipesala  i think this method can't solve the problem ,because the first transformExpressions called by sel_3q  runs unlimited and throw StackOverflowError， right？ 
Or i didn't get your idea?